### ISSUE TITLE
tests are broken

### ISSUE THREAD

**Issue by danieleades** [jhelovuo/RustDDS#42]
what is says on the tin. `cargo test` fails to compile. suspect the tests are no longer aligned with the code? did they ever work?

**Comment by jhelovuo** [comment]
That is right. Tests no longer align with the code. Some of the tests should be just fixed to be in sync with implementation code. Some others need to be redesigned altogether.

The test have worked (at least compiled), but that is several months ago.

**Comment by danieleades** [comment]
> That is right. Tests no longer align with the code. Some of the tests should be just fixed to be in sync with implementation code. Some others need to be redesigned altogether.
> 
> The test have worked (at least compiled), but that is several months ago.

i can't imagine you can refactor with any confidence without these working. Not sure i have a good enough understanding of this crate to fix the tests any further. Don't think there's too much to do though

**Comment by danieleades** [comment]
i should probably take a moment to say why i'm interesting in this!

we use DDS in my company, though we do not yet use Rust. I would like to see solid DDS support in Rust in order to be able to make a case for future adoption.
In order to make a convincing argument, any DDS implementation would have to be rock-solid and idiomatic. If there's anything I can do to help, I'm all ears. Perhaps you can tell me a little about your plans for this crate, and what level of maintenance you envision?

**Comment by jhelovuo** [comment]
This is interesting. We (Atostek Oy) have implemented RustDDS mostly to be able to talk to ROS2 from Rust natively.

What is your company doing and where are you using DDS? RustDDS of course aspires to be rock-solid and idiomatic, but still has some way to go. AFAIK it is still the leading Rust native DDS implementation.

If you would like to continue this discussion elsewhere, please reply to my email juhana.helovuo@atostek.com .

**Comment by jhelovuo** [comment]
@danieleades , If you would like to have a small excercise in RustDDS, I have a suggestion:

Replace the current examples with better ones:

* You can copy a shapes demo implementation almost directly from here: https://github.com/jhelovuo/dds-rtps/tree/master/RustDDS

* ROS2 turtle example should be completely rewritten with more or less the same functionality. The shapes demo could serve as a starting point. There should be no need for several modules, multiple threads, or internal communication channels. The current implementation is unnecessary convoluted because of earlier design mistakes (now fixed) in RustDDS and other obscure reasons.

**Comment by danieleades** [comment]
> * You can copy a shapes demo implementation almost directly from here: https://github.com/jhelovuo/dds-rtps/tree/master/RustDDS

have made a few pull requests to start tidying up the implementation

**Comment by jhelovuo** [comment]
Tests are now buildable.

**Comment by danieleades** [comment]
the tests are now buildable, but do not pass. Is that expected?

```
failures:
    dds::reader::tests::rtpsreader_handle_data
    dds::reader::tests::rtpsreader_handle_gap
    dds::reader::tests::rtpsreader_notification
    structure::dds_cache::tests::create_dds_cache
    structure::inline_qos::tests::inline_qos_key_hash

test result: FAILED. 683 passed; 5 failed; 0 ignored; 0 measured; 0 filtered out; finished in 2.24s
```
